src/Npgsql/NpgsqlConnector.cs

```diff
@@ -790,14 +790,16 @@ void SetSocketOptions(Socket socket)
             if (Settings.SocketSendBufferSize > 0)
                 socket.SendBufferSize = Settings.SocketSendBufferSize;
 
+            if (Settings.TcpKeepAlive)
+                socket.SetSocketOption(SocketOptionLevel.Socket, SocketOptionName.KeepAlive, true);
             if (Settings.TcpKeepAliveInterval > 0 && Settings.TcpKeepAliveTime == 0)
                 throw new ArgumentException("If TcpKeepAliveInterval is defined, TcpKeepAliveTime must be defined as well");
             if (Settings.TcpKeepAliveTime > 0)
             {
                 if (!PGUtil.IsWindows)
```

Comment:
Why do we test the platform if now it works on linux too?

Reply:
`TcpKeepAlive` works on Linux too, but not `TcpKeepAliveTime` (it allows you to override system defaults). The second one is done via `IOControl`, and that way isn't cross-platform. See the issue.